### ISSUE TITLE
After successful onboarding allow users to set Advance Privacy Settings and Prompt to Add Custom RPC

### ISSUE THREAD

**Issue by sethkfman** [MetaMask/metamask-mobile#5326]
### **Description**

This ticket will create the ability for users to be prompted upon wallet creation to update to a new mainnet network that uses a custom RPC. 

### **Technical Details**

-   Update the Congratulations modal with the option to link to the new Advanced Privacy Settings View
-   Add an Advanced Privacy Settings View
  -  In this view there should be an option to add Custom Ethereum Access
     -  If this user selects the Custom Access it should provide a add custom network view with Mainnet data populated except the RPC URL
  -  In this view existing Custom IPFS Gateway should be move from Advanced view
  -  Remove option for users to edit the RPC from the MetaMetrics text

### **Acceptance Criteria**

-  When users create a wallet they see a wallet creation to prompt
-  A new Advanced Privacy Settings View is displayed with (Custom Ethereum Access, Custom IPFS Gateway)
-  Users can add mainnet as a Custom Network
-  Edit the RPC from the MetaMetrics text is no longer an option
-  Translations are not required

### **References**

-   Figma: https://www.figma.com/file/4gkmBwSE9qkecCS69pwsG3/Onboarding-Redesign?node-id=582%3A25509&t=bb3M8cumJAw01kVr-0

**Comment by SaraCheikh** [comment]
Hello, here are the latest [design](https://www.figma.com/file/4gkmBwSE9qkecCS69pwsG3/Onboarding-Redesign?node-id=1415%3A60746&t=h26VUNfQFA68y70y-1) and [prototype](https://www.figma.com/proto/4gkmBwSE9qkecCS69pwsG3/Onboarding-Redesign?page-id=1345%3A57661&node-id=1415%3A60746&viewport=785%2C-145%2C0.39&scaling=min-zoom&starting-point-node-id=1415%3A60746&show-proto-sidebar=1) for this MVP. Things to note:
-There is a link to learn more about privacy in the prompt that it's supposed to direct to a new article that I'm not sure if it's ready @omnat. 
- I prototyped this MVP with updated modals for custom network confirmation and Network, but we can still use the current ones if this is out of scope for now. 
-When adding a network the current navigation defaults to` popular`, for this flow I defaulted it to [custom network](url) because it's the main action we are communicating in the prompt. 
-Now that we are showing only two settings and there is no need to group a large number of them like before, I changed the background to our current background/default for simplicity/consistency.

**Comment by Cal-L** [comment]
Notes from Design sync with Sara

Notes
- After reviewing the flow for both Mobile & Extension, we realized that there is an opportunity for feature parity between both products especially around the onboarding flow
- Both extension and mobile's onboarding has similar design - At the end of wallet creation, you can choose to either back up SRP or skip. If you choose to back up, the last screen you end up on is a celebration screen. If you choose to skip, you  land directly in the wallet.
- What's different ATM - The celebration screen is where things begin to deviate. For starters, the copies are different and the CTAs are slightly different. On the new extension design, one of the CTAs is the new privacy settings link. On mobile, the link is a "leave yourself a hint" link. None of these CTAs are platform specific and there's probably no reason for these screens to differ at all.
- There's also a question on whether or not the hint is still useful, where it's used, and why mobile has it and extension doesn't.
- Assuming we achieve parity on the celebration screen and we apply the privacy settings on both - This eliminates the need for yet another sheet pop-up post wallet creation and contains privacy settings within the onboarding flow, which is what the expected functionality implies (The fact that you don't get any data before confirming privacy settings makes it a dependency before showing the wallet)
- After that, the next thing to consider is whether or not we want to prompt existing users with the privacy settings. If the answer is no, then this is probably the extent of the proposal above. If the answer is yes, then we should try to achieve feature parity between mobile and extension. (Even the latest extension design doesn't support this)
- In conclusion - There is no reason for the flows to differ between mobile and extension since none of the features are platform specific. We can achieve parity if we unified the flows and kept privacy settings in the onboarding bucket. As an added bonus on mobile. As an added bonus, this solves the issue of the privacy settings being a dependency of showing the wallet on mobile since we would not longer show the wallet until onboarding is absolutely done.





[11:25](https://consensys.slack.com/archives/C04DBRWQH26/p1669922757107569)
There are a couple questions + potential proposal here - I can copy the notes over as well.

**Comment by Cal-L** [comment]
> After that, the next thing to consider is whether or not we want to prompt existing users with the privacy settings. If the answer is no, then this is probably the extent of the proposal above. If the answer is yes, then we should try to achieve feature parity between mobile and extension. (Even the latest extension design doesn't support this)

Seth was suggesting that this could be resolved with our existing what's new mechanism

**Comment by SaraCheikh** [comment]
Hello, I'm removing the current link to learn more about privacy from the description in advanced privacy settings prompt since we don't have an article for it yet cc:@tommasini

**Comment by sethkfman** [comment]
@SaraCheikh We are doing sprint planning for the Prod0 time. Lets us know if the designs are finalized.

**Comment by SaraCheikh** [comment]
Hi @sethkfma, I'm OOO this week, will be back next Monday. Designs for both settings and the advanced privacy settings prompt were finalized from my end, I left some questions regarding the content of settings for @omnat in the Figma file.

**Comment by SaraCheikh** [comment]
Hi @sethkfman, you can find [here a new flow](https://www.figma.com/file/4gkmBwSE9qkecCS69pwsG3/Onboarding-Redesign?node-id=2059%3A407465&t=R97TNQKXP19oTXr6-1) with the entry point of advanced configurations from the wallet creation screen, and [here a quick prototype](https://www.figma.com/proto/4gkmBwSE9qkecCS69pwsG3/Onboarding-Redesign?page-id=1345%3A57661&node-id=2058%3A69256&viewport=347%2C-35%2C0.16&scaling=scale-down&starting-point-node-id=2058%3A69256&show-proto-sidebar=1). We currently show this screen only if the user has backed up their SRP, I created an alternative confirmation screen for when users skip this step. tagged @coreyjanssen in Figma for content review

**Comment by SaraCheikh** [comment]
@sethkfman note that I'm not keeping the current "leave yourself a hint" link that we currently have. It's not a feature we have in settings and, even if we did, it wouldn't be useful when users lose their SRP since they wouldn't be able to access their wallet anyway.

**Comment by sethkfman** [comment]
@AlexJupiter @gantunesr It would be great to get your thoughts on removing this feature: https://github.com/MetaMask/metamask-mobile/issues/5326#issuecomment-1397111112

**Comment by hesterbruikman** [comment]
Tracking in #1295
